services/director-v2/src/simcore_service_director_v2/utils/dask_client_utils.py

```diff
@@ -98,35 +99,51 @@ async def _connect_with_gateway_and_create_cluster(
 ) -> DaskSubSystem:
     try:
+        logger.debug(
+            "connecting with gateway at %s with %s", f"{endpoint!r}", f"{auth_params=}"
+        )
         gateway_auth = await get_gateway_auth_from_params(auth_params)
         gateway = dask_gateway.Gateway(
             address=f"{endpoint}", auth=gateway_auth, asynchronous=True
         )
-        # if there is already a cluster that means we can re-connect to it,
-        # and IT SHALL BE the first in the list
-        cluster_reports_list = await gateway.list_clusters()
-        cluster = None
-        if cluster_reports_list:
-            assert (
-                len(cluster_reports_list) == 1
-            ), "More than 1 cluster at this location, that is unexpected!!"  # nosec
-            cluster = await gateway.connect(
-                cluster_reports_list[0].name, shutdown_on_close=False
+
+        try:
+            # if there is already a cluster that means we can re-connect to it,
+            # and IT SHALL BE the first in the list
+            cluster_reports_list = await gateway.list_clusters()
+            logger.debug(
+                "current clusters on the gateway: %s", f"{cluster_reports_list=}"
             )
-        else:
-            cluster = await gateway.new_cluster(shutdown_on_close=False)
-        assert cluster  # nosec
-        logger.info("Cluster dashboard available: %s", cluster.dashboard_link)
-        # NOTE: we scale to 1 worker as they are global
-        await cluster.adapt(active=True)
-        client = await cluster.get_client()  # type: ignore
-        assert client  # nosec
-        return DaskSubSystem(
-            client=client,
-            scheduler_id=client.scheduler_info()["id"],
-            gateway=gateway,
-            gateway_cluster=cluster,
-        )
+            cluster = None
+            if cluster_reports_list:
+                assert (
+                    len(cluster_reports_list) == 1
+                ), "More than 1 cluster at this location, that is unexpected!!"  # nosec
+                cluster = await gateway.connect(
+                    cluster_reports_list[0].name, shutdown_on_close=False
+                )
+                logger.debug("connected to %s", f"{cluster=}")
+            else:
+                cluster = await gateway.new_cluster(shutdown_on_close=False)
+                logger.debug("created %s", f"{cluster=}")
+            assert cluster  # nosec
+            logger.info("Cluster dashboard available: %s", cluster.dashboard_link)
+            # NOTE: we scale to 1 worker as they are global
+            await cluster.adapt(active=True)
+            client = await cluster.get_client()  # type: ignore
+            assert client  # nosec
+            return DaskSubSystem(
+                client=client,
+                scheduler_id=client.scheduler_info()["id"],
+                gateway=gateway,
+                gateway_cluster=cluster,
+            )
+        except Exception as exc:
```

Comment:
does it worth having a context manager for the ``gateway``?

Reply:
no way of getting the value out of the context manager when it works successfully.